doc/user_guide/io.rst

```diff
@@ -1030,6 +1032,162 @@ HDF5 file. All other keyword arguments will be passed to
 Note that the model and other secondary data artifacts linked to the signal are not
 written to the file but these can be implemented at a later stage.
 
+Nexus and general HDF
+---------------------
+
+Background
+^^^^^^^^^^
+'NeXus <https://www.nexusformat.org>'_ is a common data format orginally 
+developed by the neutron, x-ray communities. It is still being developed as 
+an international standard by scientists and programmers representing major 
+scientific facilities in order to facilitate greater cooperation in the analysis
+and visualization of data.
+Nexus uses a variety of classes to record data, values, 
+units and other experimental metadata associated with an experiment. 
+For specific types of experiments an Application Definition may exist which
+defines an agreed common layout that facilities can adhere to.
+ 
+Nexus metadata and data are stored in Hierarchical Data Format Files (HDF5) with
+a .nxs extension although standards HDF5 extensions are sometimes used.
+
+Files must use the ``.nxs``,``.hdf``,``.hdf5``,``.h5`` file extension 
+in order to use this io plugin.
+Using the ``.nxs`` extension will default to the Nexus loader 
+Using any other extension with Nexus reader will require the 
+format is specfied as Nexus in the loader 
```

Comment:
"Using any generic hdf5 extension with the Nexus reader will require the format to be specified as Nexus using the hdf5_format argument of the load function"

Related to this - please can we rename preferred_format --> hdf5_format to be more explicit?

Also a hs.load("my_data.hdf5", hdf_format="nexus")  code block example somewhere in this documentation would be helpful.

Reply:
Ok, having read more of the code I see that the user could specify any format and try to use that reader to load some file. Given that, I think it might be best to go with just "format"?